core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -728,6 +746,11 @@ private[spark] object Utils extends Logging {
     localDirs
   }
 
+  /** Used by unit tests. Do not call from other places. */
+  private[spark] def clearLocalRootDirs(): Unit = {
```

Comment:
May be we can call this function to delete the local root directory in non-yarn mode when application is exited or SparkContext is stoped.